File: 2106-FindGreatestCommonDivisorOfArray/2106-FindGreatestCommonDivisorOfArray.js
// Last updated: 10/03/2026, 10:02:58
/**
 * @param {number[]} nums
 * @return {number}
 */
var findGCD = function(nums) {
    let max=Math.max(...nums);
    let min=Math.min(...nums);
    for(let i=max;i>=1;i--){
        if(max%i==0 && min %i==0){
            return i;
        }
    }
};

File: 2106-FindGreatestCommonDivisorOfArray/2106-FindGreatestCommonDivisorOfArray.cpp
// Last updated: 03/08/2026, 10:08:04
class Solution {
public:
    int findGCD(vector<int>& nums) {

        int mn = nums[0];
        int mx = nums[0];

        for (int i = 0; i < nums.size(); i++) {
            mn = min(mn, nums[i]);
            mx = max(mx, nums[i]);
        }

        return gcd(mn, mx);
    }
};